### ISSUE TITLE
[Question] Getting it to work with DNSDynamic.org

### ISSUE THREAD

**Issue by Zero3K** [synappnz/ipright#1]
How do I do that?

**Comment by synappnz** [comment]
The updates to DNS Park are "hard coded". I've taken a look at the DNSDynamic web site and it doesn't look like it would be too difficult to implement their updating. Is this something that you need or is this just a general question?

**Comment by Zero3K** [comment]
Its something that I need . Also, it would be nice if it could also support DSLReport.com's Monitoring service. The url used to update its IP address that is pinged is http://username:password@dslreports.com/nic?action=edit&started=1&host_id
=NNNNNN&myip=A.B.C.D.

**Comment by synappnz** [comment]
Could you please send me a link to the dslreports.com monitoring service information.

**Comment by Zero3K** [comment]
Here's the link to it: http://www.dslreports.com/faq/monitorfaq

**Comment by synappnz** [comment]
I've created a new branch, dnsdynamic. The IpRight.exe in that branch should work with DNSdynamic. Could you replace the IpRight.exe that you're currently using with the one on the dnsdynamic branch and let me know how you get on. The DNS Park updates have been removed in this branch and it only supports one hostname. All IP address update error codes are currently not handled, I will fix this later. The API example at DNSdynamic shows only updating one hostname at a time but the IP Right code currently allows updating multiple hostnames on a single update request rather than multiple hostnames, each on a seperate request. The code will have to be restructured to handle updating multiple hostnames per account, one hostname per request. If you only have one hostname, please let me know (as I won't restructure the code unless actually necessary). I'll take a look at dnsreports.com next.

**Comment by Zero3K** [comment]
So, there's no way to make it where there's a drop down box with a list of popular DDNS services which will allow the user to use IP Right with one of them? (Basically, DNS Park, DNSDynamic and DSLReports Monitoring could be a couple of the options). Regarding the DNSDynamic support that's added, the log states "Check IP address socket disconnected". The contents of the XML it uses are as follows:

< IpRight >
  < Log Enabled="1" Methods="1" Options="1" Checks="1" Updates="1" />
  < Check Enabled="1" Tick="600" Internet="1" Server="dnsdynamic.org" Port="80" Username="" Password="" Get="" ExcludeIpAddresses="0.0.0.0" />
  < Account Disabled="0" Username="Zero3K@myopera.com" Password="50240220">
    < Host Disabled="0" Name="zero3k.dnsdynamic.com" IpAddress="" Updated="" ResultCode="" />
  < /Account>
< /IpRight >

**Comment by synappnz** [comment]
Yes, I'll have all three services supported together once I have the DNSdynamic code working in it's own branch. I'm restructuring the code in that branch to allow for multiple services. Once complete, I'll merge the branch back into master.

I notice that there's not an IP Address in the IpAddress attribute of the Host element. It looks like you need a few changes to the Check section. Change Interval in Seconds to 60 and change Server to myip.dnsdynamic.org. Your IP address will then be correctly detected and updated.

**Comment by Zero3K** [comment]
Why 60 seconds? I think that 600 seconds (10 minutes) is long enough to check to see if the IP address was changed or not.

**Comment by Zero3K** [comment]
I also think that some of the services listed on http://dnslookup.me/dynamic-dns/ should be supported (if you feel like doing so).

**Comment by Zero3K** [comment]
So, I changed the options (except the Checking Interval) like you wrote and it works.

EDIT: The log states that the host I have specified is out of date even though it isn't.

**Comment by synappnz** [comment]
The first time IP Right checks the IP address and updates a new host, that host's IP address in the config file will be blank so it will update it. The result code will indicate that there is no change so IP Right should update the IpAddress attribute for that host with the IP address that it has just detected. This should only happen once for new hosts. If it happens any more than once then the update client may be blocked. Could you let me know if the IPAddress attribute has your current IP address.

I managed to get another four hours of code restructuring in tonight which has left IP Right in a broken state so I haven't committed  the code. It should be ready to go for DNS Park, DNSdynamic and DSLReport next time I get a change to work on it.

**Comment by Zero3K** [comment]
The IPAddress attribute has no value in it even after a few hours of it running.

**Comment by synappnz** [comment]
Could you email me the log please: stacey.richards@ird.co.nz

**Comment by Zero3K** [comment]
Its working now. The IPAddress attribute is set to my current IP address.
